pandas/core/arrays/interval.py

```diff
@@ -561,6 +563,106 @@ def from_tuples(
 
         return cls.from_arrays(left, right, closed, copy=False, dtype=dtype)
 
+    _interval_shared_docs["from_strings"] = textwrap.dedent(
+        """
+        Construct from string representations of the left and right bounds.
+
+        Parameters
+        ----------
+        data : array-like (1-dimensional)
+            Strings representing the Interval's to parse.
+        copy : bool, default False
+            Copy the data.
+        dtype : dtype, optional
+            If None, dtype will be inferred.
+
+        Returns
+        -------
+        %(klass)s
+
+        Raises
+        ------
+        ValueError
+            When a string cannot be parsed as an Interval
+            When the dtype of the string cannot be parsed as either float,
+            Timestamp or Timedelta
+
+        See Also
+        --------
+        interval_range : Function to create a fixed frequency IntervalIndex.
+        %(klass)s.from_breaks : Construct an %(klass)s from an array of
+            splits.
+        %(klass)s.from_tuples : Construct an %(klass)s from an
+            array-like of tuples.
+
+        %(examples)s\
+        """
+    )
+
+    @classmethod
+    @Appender(
+        _interval_shared_docs["from_strings"]
+        % {
+            "klass": "IntervalIndex",
+            "examples": textwrap.dedent(
+                """\
+        Examples
+        --------
+        >>> pd.IntervalIndex.from_strings(["(0, 1]", "(1, 2]"])
+        IntervalIndex([(0, 1], (1, 2]],
+                       dtype='interval[int64, right]')
+        """
+            ),
+        }
+    )
+    def from_strings(
+        cls: type[IntervalArrayT],
+        data: Sequence[str],
+    ) -> IntervalArrayT:
+        # These need to be imported here to avoid circular dependencies.
+        from pandas.core.tools.datetimes import to_datetime
+        from pandas.core.tools.timedeltas import to_timedelta
+
+        intervals: list[Interval] = []
+        for string in data:
+
+            # Try to match "(left, right]" where 'left' and 'right' are breaks.
+            breaks_match = re.match(r"\(.*,.*]", string)
+            # Raise ValueError if no match was found.
+            if breaks_match is None:
+                raise ValueError(
+                    "Could not find opening '(' and closing ']' "
+                    f"brackets in string: '{string}'"
+                )
+
+            # Try to split 'left' and 'right' based on a comma and a space.
+            breaks = breaks_match.string[1:-1].split(", ", 1)
+
+            if len(breaks) != 2:
+                raise ValueError(
+                    f"Delimiter ', ' (comma + space) not found in string: {string}"
+                )
+
+            conversions: list[Callable] = [int, float, to_datetime, to_timedelta]
+            # Try to parse the breaks first as floats, then datetime, then timedelta.
```

Comment:
hmm i am not a big fan of this, can we make the dtype strict e.g. `interval[float]` would completely solve this issue (forcing the user to do this)

Reply:
the problem here is that there IS ambiguity.

Reply:
The case that this string parsing functionality is relevant (and the only one I see for now) is reading from text files. The other parts of pandas infer types dynamically, so wouldn't it be best to do this by default here as well?

In 162045a, I added an optional `dtype` argument. If specified (not `None`), it will parse string representations first and then rely on `IntervalArray.from_arrays()` to (potentially) perform the more exact conversion. The `IntervalArray.from_arrays()` method unfortunately doesn't parse string representations of numeric or datetime-like values, so the `IntervalArray.from_strings()` method has to implement it in some way:
```python
In [1]: import pandas as pd

In [2]: pd.arrays.IntervalArray.from_arrays(["0", "1", "2"], ["0", "1", "2"])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-677073bd0961> in <module>
----> 1 pd.arrays.IntervalArray.from_arrays(["0", "1", "2"], ["0", "1", "2"])

~/Projects/pandas/pandas/core/arrays/interval.py in from_arrays(cls, left, right, closed, copy, dtype)
    479         right = _maybe_convert_platform_interval(right)
    480
--> 481         return cls._simple_new(
    482             left, right, closed, copy=copy, dtype=dtype, verify_integrity=True
    483         )

~/Projects/pandas/pandas/core/arrays/interval.py in _simple_new(cls, left, right, closed, copy, dtype, verify_integrity)
    293                 "for IntervalArray"
    294             )
--> 295             raise TypeError(msg)
    296         elif isinstance(left, ABCPeriodIndex):
    297             msg = "Period dtypes are not supported, use a PeriodIndex instead"

TypeError: category, object, and string subtypes are not supported for IntervalArray
```